pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go

```diff
@@ -1904,36 +1916,52 @@ type replicaSendStream struct {
 			tokenWatcherHandle         SendTokenWatcherHandle
 			deductedForSchedulerTokens kvflowcontrol.Tokens
 		}
+		// TODO(sumeer): remove closed. Whenever a replicaSendStream is closed it
+		// is also no longer referenced by replicaState. The only motivation for
+		// closed is that replicaSendStream.Notify calls directly into
+		// replicaSendStream. But closing a send stream also sets
+		// replicaSendStream.mu.sendQueue.tokenWatcherHandle to empty, and Notify
+		// is already a noop in that case. So this field serves no useful purpose.
 		closed bool
 	}
+	// Fields that are read/written while holding raftMu.
+	raftMu struct {
+		// tracker contains entries that have been sent, and have had send-tokens
+		// deducted (and will have had eval-tokens deducted iff index >=
+		// nextRaftIndexInitial).
```

Comment:
Aren't eval tokens deducted first? Should "will have had" be "have had" or some form of the past?
Could you make the order of things clearer in this comment?

Reply:
> Aren't eval tokens deducted first?

Correct.

> Should "will have had" be "have had" or some form of the past? Could you make the order of things clearer in this comment?

Yep -- I'll do so in the next PR. I want to merge this now given CI was green before the last round of changes, and I only made superficial changes in that last round.